### PR TITLE
[Feature] Merge chat branch into develop

### PR DESCRIPTION
**Pull request by emlynmac**
## Purpose
This PR is to merge the chat feature branch into develop branch to simplify the development process moving forwards.

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What to Check
All chat and calling functionality should be unaffected. 
CI will now do both chat and calling for all PR commits

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/DiagnosticConfig.swift [hunk 1]
```diff
@@ -29,3 +29,13 @@ struct DiagnosticConfig {
     }
 
 }
+
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/DiagnosticConfig.swift:32]: I think this code should be removed as there should be a separate file for this info (DebugInfo)

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Manager/DiagnosticsManager.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import Combine
+import Foundation
+
+protocol DiagnosticsManagerProtocol {
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/Manager/DiagnosticsManager.swift:9]: looks like copy of DiagnosticsManager file. Also, `getDiagnosticsInfo` func isn't correct

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/ViewComponents/SharingActivity/DiagnosticsSharingActivityViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import Foundation
+
+class DiagnosticsSharingActivityViewModel {
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/ViewComponents/SharingActivity/DiagnosticsSharingActivityViewModel.swift:8]: Is it a copy as well?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Mocking/DiagnosticsManagerMocking.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import Foundation
+@testable import AzureCommunicationUICalling
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Tests/Mocking/DiagnosticsManagerMocking.swift:7]: Is it a copy as well?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling.podspec [hunk 5]
```diff
@@ -5,21 +5,22 @@ Pod::Spec.new do |spec|
...
   spec.platform             = :ios, '14.0'
 
-  spec.source_files         = 'AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/**/*.swift'
+  spec.source_files         =  'AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/**/*.swift', 'AzureCommunicationUI/sdk/AzureCommunicationUICommon/Sources/AzureCommunicationUICommon/*.swift', 'AzureCommunicationUI/sdk/common/**/Sources/**/*.swift'
   spec.resources            = 'AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/**/*.{xcassets,strings}'
   spec.pod_target_xcconfig  = { "EXCLUDED_ARCHS[sdk=iphonesimulator*]": "arm64", "ENABLE_BITCODE": "NO"}
   spec.info_plist           = {'UILibrarySemVersion' => "#{spec.version}"}
 
   spec.dependency             'AzureCommunicationCalling', '2.2.1'
-  spec.dependency             'MicrosoftFluentUI/Avatar_ios', '0.3.9'
```
Comment (palatter) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling.podspec:20]: Wasn't develop already at 0.8.9?

File: README.md [hunk 6]
```diff
@@ -1,11 +1,7 @@
-![Hero Image](https://github.com/Azure/communication-ui-library-ios/blob/main/docs/images/mobile-ui-library-hero-image.png?raw=true)
+![Hero Image](/docs/images/mobile-ui-library-hero-image.png?raw=true)
 
 # Azure Communication UI Mobile Library for iOS
```
Comment (palatter) [inline README.md:4]: Did we mean to remove these?

File: README.md [hunk 7]
```diff
@@ -19,50 +15,34 @@ Get started with Azure Communication Services by using the UI Library to integra
...
 
-CocoaPods is a dependency manager for Cocoa projects. To set up with CocoaPods visit their [Getting Started Guide](https://guides.cocoapods.org/using/getting-started.html). To integrate UI Mobile Library into your Xcode project using CocoaPods, specify it in your `Podfile`:
+Azure Communication Service Mobile UI currently offers the following libraries:
 
-```ruby
-pod 'AzureCommunicationUICalling', '1.1.0'
-```
+### [Calling](AzureCommunicationUI/sdk/AzureCommunicationUICalling)
 
-#### Manual Installation
```
Comment (palatter) [inline README.md:38]: Same here?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling.podspec (hunk 5 above)
Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling.podspec:20]: Maybe not for the podspec?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling.podspec [hunk 8]
```diff
@@ -5,21 +5,22 @@ Pod::Spec.new do |spec|
   spec.homepage             = "https://github.com/Azure/communication-ui-library-ios"
   spec.license              = { :type => 'MIT' }
   spec.author               = 'Microsoft'
-  spec.source               = { :git => 'https://github.com/Azure/communication-ui-library-ios.git', :tag => 'AzureCommunicationUICalling_1.1.0' }
+  spec.source               = { :git => 'https://github.com/Azure/communication-ui-library-ios.git', :branch => 'feature/chat' }
```
[the remark this answers is not in the document]
Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling.podspec:8]: Should be unchanged; reverting.
Also updating the chat pod spec.

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/DiagnosticConfig.swift (hunk 1 above)
Comment (emlynmac) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/DiagnosticConfig.swift:32]: It is used in the CallConfiguration class: https://github.com/Azure/communication-ui-library-ios/blob/develop/AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallConfiguration.swift

File: eng/pipelines/ci.yml [hunk 9]
```diff
@@ -10,10 +10,5 @@ pr:
   - release/*
   - feature/*
```
[the remark this answers is not in the document]
Comment (emlynmac) [reply eng/pipelines/ci.yml:12]: It is done already; but good thing to check :)

File: README.md (hunk 6 above)
Comment (emlynmac) [reply README.md:4]: Yes, see https://github.com/Azure/communication-ui-library-ios/pull/491

File: README.md (hunk 7 above)
Comment (emlynmac) [reply README.md:38]: See https://github.com/Azure/communication-ui-library-ios/pull/491

**Comment by emlynmac** [comment]
> hey would you mind to rebase merge-chat-develop, I just merged a bug fix to chat

Can you merge the change into the merge-chat-dev branch please?

**Review by Leah-Xia-Microsoft — APPROVED** [review]

File: eng/pipelines/ci.yml [hunk 10]
```diff
@@ -10,10 +10,5 @@ pr:
...
```
Comment (jpeng-ms) [inline eng/pipelines/ci.yml:11]: it seems this PR only triggers build for chat, should we have two builds instead?

Comment (emlynmac) [reply eng/pipelines/ci.yml:11]: We need only one CI build, which builds everything and does code coverage. Have a look at the checks section to see what is happening. I'm happy to explain the CI YAML again, but we should do that offline.

Comment (jpeng-ms) [reply eng/pipelines/ci.yml:11]: ah you are right, somehow missed it, thanks!

File: README.md (hunk 6 above)
Comment (jpeng-ms) [reply README.md:4]: yes the image path should be updated to relative path instead of being hard coded to pointing main branch

**Review by palatter — COMMENTED** [review]
Looks good.

**Review by palatter — APPROVED** [review]

**Pull request merged**